### ISSUE TITLE
Issue when launching the python debug mode in Azure function

### ISSUE THREAD

**Issue by georged4s** [microsoft/vscode-azurefunctions#2041]
### **System Specs:**
**OS: Ubuntu 18.04**
**Vscode Version: 1.44.2 (user setup)
Commit: ff915844119ce9485abfe8aa9076ec76b5300ddd
Date: 2020-04-16T16:36:23.138Z
Electron: 7.1.11
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.18363**
### **Steps to Reproduce**:
*  follow the official [documentation](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-azure-function-azure-cli?tabs=bash%2Cbrowser&pivots=programming-language-python) to create a new azure function project (except in the first command use `python3 -m venv .venv` instead)
* Reload Window (Ctrl + R or Developer: Reload Window in the command palette)
* Allow vscode to create its configuration files for the azure function project by clicking on "Yes" when you see the below pops up
![image](https://user-images.githubusercontent.com/57380498/79738848-ac4b8f00-8340-11ea-8502-3b1d54a3017c.png)
* Open the `__init__.py` file located under `LocalFunctionProj/HttpExample` and add a line to import a package not installed (I have used "pandas") under the your root python site-packages folder (in my case it is /home/<my_username>/.local/lib/python3.6/site-packages)
* Add this package to the `requirements.txt` under `LocalFunctionProj`
![image](https://user-images.githubusercontent.com/57380498/79740686-6e9c3580-8343-11ea-8001-648c5c8f56c5.png)
* Run `source .venv/bin/activate` to activate the virtual environment
* Run `pip install -r ./LocalFunctionProj/requirements.txt` to install all dependencies
* Press `F5` key to enter into the debug mode
* Open up this link in your browser http://localhost:7071/api/HttpExample?name=Functions and see that it should throw an `ModuleNotFoundError` error about pandas not being imported
![image](https://user-images.githubusercontent.com/57380498/79741203-32b5a000-8344-11ea-982e-6cd0ab6add9b.png)
### **Diagnosis**
After some researches I came to notices that it is not activating the virtual environment when using the debug mode. Here is how I found it out
* Delete `import pandas` and add `import sys`
* Add `logging.info("Python executable path is: {}".format(sys.executable))` to log the python executable directory anywhere in the `__init__.py` before the function ends
* Launch the debug mode by pressing `F5`
* Refresh http://localhost:7071/api/HttpExample?name=Functions and it prints out this
![image](https://user-images.githubusercontent.com/57380498/79742303-f125f480-8345-11ea-833e-c5859d43d70a.png)

### **Solution**
At the end, I solved it by adding `"azureFunctions.pythonVenv": "${workspaceFolder}/.venv"` in the `.vscode/settings.json` file
![image](https://user-images.githubusercontent.com/57380498/79742667-8923de00-8346-11ea-942f-658623631fcf.png)
Btw, I don't think this value `"python.pythonPath": ".venv/bin/python3"` is used at all as even I have removed it completed it doesn't seem to impact the usability of neither the debug mode nor of `func start`.

<br><br>
Here are my configuration files:
`extensions.json`
`
{
  "recommendations": [
    "ms-azuretools.vscode-azurefunctions",
    "ms-python.python"
  ]
}
`<br>
`launch.json`
`
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Attach to Python Functions",
      "type": "python",
      "request": "attach",
      "port": 9091,
      "preLaunchTask": "func: host start"
    }
  ]
}
`<br>
`settings.json`
`
{
    "azureFunctions.pythonVenv": "${workspaceFolder}/.venv",
    "azureFunctions.deploySubpath": "LocalFunctionProj",
    "azureFunctions.scmDoBuildDuringDeployment": true,
    "azureFunctions.projectLanguage": "Python",
    "azureFunctions.projectRuntime": "~2",
    "debug.internalConsoleOptions": "neverOpen",
    "python.pythonPath": ".venv/bin/python3"
}
`<br>
`tasks.json`
`
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "func",
      "command": "host start",
      "problemMatcher": "$func-watch",
      "isBackground": true,
      "options": {
        "cwd": "${workspaceFolder}/LocalFunctionProj"
      }
    }
  ]
}
`

**Comment by EricJizbaMSFT** [comment]
When you hit this dialog:
![image](https://user-images.githubusercontent.com/57380498/79738848-ac4b8f00-8340-11ea-8502-3b1d54a3017c.png)

It was supposed to automatically set "azureFunctions.pythonVenv" to ".venv". Did the venv already exist when you hit "Yes"? And what was the value of this setting immediately after?

**Comment by georged4s** [comment]
> When you hit this dialog:
> ![image](https://user-images.githubusercontent.com/57380498/79738848-ac4b8f00-8340-11ea-8502-3b1d54a3017c.png)
> 
> It was supposed to automatically set "azureFunctions.pythonVenv" to ".venv". Did the venv already exist when you hit "Yes"? And what was the value of this setting immediately after?

Yes, the .venv already exists when this msg pops up and my folder structure looks like this.
![image](https://user-images.githubusercontent.com/57380498/79810559-b019f880-83b6-11ea-88d7-7a951e15ec65.png)
The the original values of settings.json has everything except the `azureFunctions.pythonVenv`
I have also tried using `"azureFunctions.pythonVenv": ".venv"` as well but then I got this error.
![image](https://user-images.githubusercontent.com/57380498/79810464-75b05b80-83b6-11ea-8761-cea2a7697203.png)

**Comment by georged4s** [comment]
I am guessing that the .venv should sit inside the project folder because it does make sense to merge both the .vscode, but if I follow the doc I end up having it sitting outside

**Comment by EricJizbaMSFT** [comment]
Oh interesting - yeah probably a bug in our code related to your file structure. I think we've got enough info to investigate. In the meantime, sounds like you're already unblocked

**Comment by georged4s** [comment]
yes i am fine for now and thanks for your attentions. I think you get the idea but I want to clarify a bit more just in case. It wasn't because that the `azureFunctions.pythonVenv` was added to the wrong folder but instead it was not be added at all.

**Comment by EricJizbaMSFT** [comment]
Note to self: [Others](https://github.com/anthonychu/functions-python-durable-image-classifier) have set "azureFunctions.pythonVenv" to "../.venv" as a workaround. Need to make sure I try out any fix with projects that already have both workarounds and make sure I don't break those people
